homeassistant/components/viam/__init__.py

```diff
@@ -0,0 +1,367 @@
+        vol.Required(SERVICE_COMPONENT_NAME): vol.All(str),
+        vol.Required(SERVICE_COMPONENT_TYPE, default="sensor"): vol.All(str),
+    }
+)
+
+IMAGE_SERVICE_FIELDS = {
+    vol.Optional(SERVICE_FILEPATH): vol.All(str, vol.IsFile),
+    vol.Optional(SERVICE_CAMERA): vol.All(str),
+}
+VISION_SERVICE_FIELDS = {
+    vol.Optional(SERVICE_CONFIDENCE, default="0.6"): vol.All(
+        str, vol.Coerce(float), vol.Range(min=0, max=1)
+    ),
+    vol.Optional(SERVICE_ROBOT_ADDRESS): vol.All(str),
+    vol.Optional(SERVICE_ROBOT_SECRET): vol.All(str),
+}
+
+CAPTURE_IMAGE_SERVICE_SCHEMA = vol.Schema(
+    {
+        **IMAGE_SERVICE_FIELDS,
+        vol.Optional(SERVICE_FILE_NAME, default="camera"): vol.All(str),
+        vol.Optional(SERVICE_COMPONENT_NAME): vol.All(str),
+    }
+)
+
+CLASSIFICATION_SERVICE_SCHEMA = vol.Schema(
+    {
+        **IMAGE_SERVICE_FIELDS,
+        **VISION_SERVICE_FIELDS,
+        vol.Required(SERVICE_CLASSIFIER_NAME): vol.All(str),
+        vol.Optional(SERVICE_COUNT, default="2"): vol.All(str, vol.Coerce(int)),
+    }
+)
+
+DETECTIONS_SERVICE_SCHEMA = vol.Schema(
+    {
+        **IMAGE_SERVICE_FIELDS,
+        **VISION_SERVICE_FIELDS,
+        vol.Required(SERVICE_DETECTOR_NAME): vol.All(str),
+    }
+)
+
+
+def _fetch_image(filepath: str | None):
+    if filepath is None:
+        return None
+    return Image.open(filepath)
+
+
+class ViamManager:
+    """Manage Viam client and entry data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        client: ViamClient,
+        entry_id: str,
+        data: dict[str, Any],
+    ) -> None:
+        """Store initialized client and user input data."""
+        self.hass = hass
+        self.client = client
+        self.data = data
+        self.entry_id = entry_id
+
+    def register_services(self) -> None:
+        """Register all available services provided by the integration."""
+        self.hass.services.async_register(
+            DOMAIN, "capture_data", self.capture_data, DATA_CAPTURE_SERVICE_SCHEMA
+        )
+        self.hass.services.async_register(
+            DOMAIN, "capture_image", self.capture_image, CAPTURE_IMAGE_SERVICE_SCHEMA
+        )
+        self.hass.services.async_register(
+            DOMAIN,
+            "get_classifications",
+            self.get_classifications,
+            CLASSIFICATION_SERVICE_SCHEMA,
+            supports_response=SupportsResponse.ONLY,
+        )
+        self.hass.services.async_register(
+            DOMAIN,
+            "get_detections",
+            self.get_detections,
+            DETECTIONS_SERVICE_SCHEMA,
+            supports_response=SupportsResponse.ONLY,
+        )
+
+    def unload(self) -> None:
+        """Clean up any open clients."""
+        self.hass.services.remove(DOMAIN, "capture_data")
+        self.hass.services.remove(DOMAIN, "capture_image")
+        self.hass.services.remove(DOMAIN, "get_classifications")
+        self.hass.services.remove(DOMAIN, "get_detections")
```

Comment:
```suggestion
```

Services should not be removed

Reply:
I'll update that section. Can you help me understand why services don't need to be unregistered when unloading integrations?

Reply:
Both questions are answered in https://github.com/home-assistant/core/pull/105929#discussion_r1433754113